### ISSUE TITLE
[Listbox] Screen reader doesn't announce state of ListboxOption

### ISSUE THREAD

**Issue by zpeterson** [reach/reach-ui#745]
### Current Behavior

Screen reader does not announce the state as selected/unselected when user navigates to a ListboxOption.

### Expected behavior

Screen reader should announce the state as selected/unselected when user navigates to a ListboxOption.

### Reproducible example

https://reach.tech/listbox

### Suggested solution(s)

Add an `aria-selected` attribute to ListboxOption.

### Additional context

N/A

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Reach Package | @reach/listbox | 0.11.0 |
| React  |  | 16.12.0 |
| Browser | Chrome | 88.0.4324.150 |
| Assistive tech   | NVDA | 2020.3 |
| Node |  | 10.13.0 |
| npm/yarn | npm | 6.4.1 |
| Operating System | Windows 10 | 19042.746 |

**Comment by indiesquidge** [comment]
@zpeterson, thanks for letting us know about this. I don't have a Windows laptop with me right now so I can't play around with this in person, but the ListboxOption does already programmatically set an `aria-selected` attribute as [seen here][1]. Could there be another reason your screenreader is not reading out the selected state?

[1]: https://github.com/reach/reach-ui/blob/881b6ee3496c4b5b868f5ca1f706af63ea48114e/packages/listbox/src/index.tsx#L1058

**Comment by mgostisha** [comment]
Hey there, was looking into this one this morning. I'm seeing something similar on MacOS + VoiceOver and attached a screen recording below. It looks like the `aria-activedescendant` and `aria-selected` are being set properly according to the spec for roles of `listbox` and `option` here: https://www.w3.org/TR/wai-aria-1.1/#option.

Aside from that, it looks like there's something potentially overriding the announcement, you can see VoiceOver flash `Option (X of 5)` and then quickly it's overwritten by `Option`. Going to keep looking myself, but figured I'd post here in case anyone's seen this before.

https://user-images.githubusercontent.com/6979064/112319006-a01c3000-8c7b-11eb-986d-59287b11f11d.mov

**Comment by mgostisha** [comment]
The above VoiceOver changing issue happens on Edge and Chrome, but not Firefox, so i'm thinking that may be a browser-level issue.

**Comment by chaance** [comment]
Had some time to explore this a bit today, and I think even though we are following the suggested ARIA implementation for `aria-activedescendant`/`aria-selected`, in practice I'm not sure it's the ideal choice at this point. Other implementations I've seen that use a roving `tabindex` to track the navigation value and `aria-selected` to point to the actual selection seem to provide better context in both VoiceOver and NVDA.

**Comment by chaance** [comment]
Another thing to note, the ARIA example of a dropdown listbox actually does change the value of the listbox when navigating via keyboard, so perhaps if this were the desired behavior `aria-selected`/`aria-activedescendant` would make sense to point to the navigated value since it's the same as the selected value. We actively chose not to follow that pattern because that's not how native selects work. https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-collapsible.html
